src/util/string.cpp

```diff
@@ -13,4 +13,42 @@ void ReplaceAll(std::string& in_out, const std::string& search, const std::strin
     if (search.empty()) return;
     in_out = std::regex_replace(in_out, std::regex(search), substitute);
 }
+
+LineReader::LineReader(std::span<const std::byte> buffer, size_t max_read)
+    : start(buffer.begin()), end(buffer.end()), max_read(max_read), it(buffer.begin()) {}
+
+std::optional<std::string> LineReader::ReadLine()
+{
+    if (it == end) {
+        return std::nullopt;
+    }
+
+    auto line_start = it;
+    std::string line{};
+    while (it != end) {
+        char c = static_cast<char>(*it);
+        line += c;
+        ++it;
+        if (c == '\n') break;
+        if ((size_t)std::distance(line_start, it) >= max_read) throw std::runtime_error("max_read exceeded by LineReader");
+    }
+
+    line = TrimString(line); // delete trailing \r and/or \n
```

Comment:
Can we return a `string_view` to the internal buffer?
(preventing allocation & copy)